superset/connectors/sqla/models.py

```diff
@@ -231,6 +231,8 @@ def db_engine_spec(self) -> Type[BaseEngineSpec]:
     @property
     def type_generic(self) -> Optional[utils.GenericDataType]:
         column_spec = self.db_engine_spec.get_column_spec(self.type)
+        if self.is_dttm:
+            return GenericDataType.TEMPORAL
```

Comment:
we can probably perform this check before `get_column_spec` on the row above (I believe it's slightly more expensive than the if)

Reply:
nice catch, fixed.